### PR TITLE
Implement RemoveLegacyServiceAccount function

### PR DESCRIPTION
**Pull request by gitlawr**

### CODE REVIEW

**Comment by guangbochen** [comment]
LGTM

**Review by guangbochen — COMMENTED** [review]
need to add timeout for the waitAliyunCluster funnc

**Comment by gitlawr** [comment]
Updated

File: aliyun_driver.go [hunk 1]
```diff
@@ -540,6 +545,10 @@ func (d *Driver) waitAliyunCluster(ctx context.Context, svc *cs.Client, state *s
 			return fmt.Errorf("aliyun failed to provision cluster: %s", status)
 		}
 		time.Sleep(time.Second * 15)
+
+		if startTime.Add(time.Duration(timeoutMins) * time.Minute).Before(time.Now()) {
+			return fmt.Errorf("timeout waiting cluster %q to be running", state.Name)
+		}
```
Comment (guangbochen) [inline aliyun_driver.go:554]: Would the logic below for timeout will make the code more readable? Thanks.
``` golang
	timeout := time.After(30 * time.Minute)
	tick := time.Tick(15 * time.Second)
	// Keep trying until we're timed out or got a result or got an error
	for {
		select {
		// Got a timeout! fail with a timeout error
		case <-timeout:
			return fmt.Errorf("timedout waiting cluster ....")
		// Got a tick, we should check on cluster state and return
		case <-tick:
			cluster, err := getCluster(svc, state)
                         // return something
		}
	}
```

Comment (gitlawr) [reply aliyun_driver.go:554]: Updated

File: aliyun_driver.go [hunk 2]
```diff
@@ -778,13 +791,50 @@ func (d *Driver) PostCheck(ctx context.Context, info *types.ClusterInfo) (*types
 	return info, nil
 }
 
+func getClientset(info *types.ClusterInfo) (*kubernetes.Clientset, error) {
```
Comment (nathan-jenan-rancher) [inline aliyun_driver.go:747]: This func shares common logic with `PostCheck`, please refactor that (I would probably have `PostCheck` call this func).

**Comment by nathan-jenan-rancher** [comment]
LGTM

**Pull request merged**